src/github.com/matrix-org/go-neb/services/github/github.go

```diff
@@ -167,9 +179,90 @@ func (s *githubService) Register() error {
 	if realm.Type() != "github" {
 		return fmt.Errorf("Realm is of type '%s', not 'github'", realm.Type())
 	}
+
+	// In order to register the GH service, you must have authed with GH.
+	cli := s.githubClientFor(s.ClientUserID, false)
+	if cli == nil {
+		return fmt.Errorf("User %s does not have a Github auth session.", s.ClientUserID)
+	}
+
 	return nil
 }
 
+func (s *githubService) PostRegister(oldService types.Service) {
+	cli := s.githubClientFor(s.ClientUserID, false)
+	if cli == nil {
+		log.Errorf("PostRegister: %s does not have a github session", s.ClientUserID)
+		return
+	}
+	old, ok := oldService.(*githubService)
+	if !ok {
+		log.Error("PostRegister: Provided old service is not of type GithubService")
+		return
+	}
+
+	// remove any existing webhooks this service created on the user's behalf
+	modifyWebhooks(old, cli, true)
+
+	// make new webhooks according to service config
+	modifyWebhooks(s, cli, false)
```

Comment:
I need to delete old hooks before adding new hooks else I might delete a hook I just added. I can only delete them in `PostRegister`. That's why I add hooks here rather than in `Register()`.